crates/oxc_linter/src/rules/unicorn/no_useless_switch_case.rs

```diff
@@ -66,7 +66,7 @@ impl Rule for NoUselessSwitchCase {
         let default_case = default_cases[0];
 
         // Check if the `default` case is the last case
-        if std::ptr::from_ref(default_case) != std::ptr::from_ref(cases.last().unwrap()) {
+        if !std::ptr::eq(default_case, cases.last().unwrap()) {
```

Comment:
not sure why i didn't write this originally, but i htink comparing spans is much clearer than checking if the pointers point to the same thing

Reply:
In general we shouldn't use `Span`s as unique identifiers. In the linter, it's probably OK, as the AST has come straight from the parser. But if for some reason we wanted to lint code that's come out of the transformer, `Span`s can no longer be trusted to be unique. e.g. `x ?? y` is transformed to `x !== null && x !== void 0 ? x : y` (and all those `x`s have the same `Span`).

Comparing `Span`s is also more error-prone than comparing pointers, as `ptr::eq` will error if the 2 pointers don't have the same type, whereas any 2 `Span`s can be compared.

So personally I'm not against comparing pointers, it's purely the verbose syntax `std::ptr::from_ref(x) != std::ptr::from_ref(y)` that I'm not keen on.

Reply:
Should these be replaced with `ContentEq`? I did similar stuff just now https://github.com/oxc-project/oxc/pull/5648/files

Reply:
i'm not really against either, but from a developer point of view, i think comparing spans is a little more user friendly?
to be clear, don't let this block merging this

Reply:
`NoDuplicateCase` has to compare whether expressions are exactly the same.

> to be clear, don't let this block merging this. 🙂

Agree.

Reply:
does `NoDuplicateCase` have different spans?

Reply:
i implemented https://github.com/oxc-project/oxc/blob/59abf27d9563165883491e7ccaa89f0d78433559/crates/oxc_linter/src/utils/unicorn.rs#L150 a while ago to allow for catching stuff like is `foo as bar` the same as `foo`. in mose cases we would want to use that?